app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -165,9 +175,10 @@ export class AddExistingRepository extends React.Component<
   }
 
   private onPathChanged = async (path: string) => {
-    const isRepository = await isGitRepository(path)
-
-    this.setState({ path, isRepository })
+    if (this.state.path !== path) {
+      this.setState({ path, isRepository: false })
+      this.validatePath(path)
```

Comment:
```suggestion
      this.updatePath(path)
```

Reply:
@niik not super important but this small cleanup looks 👍  too, right?

Reply:
Ah yup!